src/Magnum/Platform/AndroidApplication.cpp

```diff
@@ -176,6 +178,71 @@ bool AndroidApplication::tryCreate(const Configuration& configuration, const GLC
     return _context->tryCreate(glConfiguration);
 }
 
+bool AndroidApplication::reactivateContext() {
+    /* Choose config */
+    const EGLint configAttributes[] = {
+        EGL_SURFACE_TYPE, EGL_WINDOW_BIT,
+        EGL_RED_SIZE, _glConfiguration.colorBufferSize().r(),
+        EGL_GREEN_SIZE, _glConfiguration.colorBufferSize().g(),
+        EGL_BLUE_SIZE, _glConfiguration.colorBufferSize().b(),
+        EGL_ALPHA_SIZE, _glConfiguration.colorBufferSize().a(),
+        EGL_DEPTH_SIZE, _glConfiguration.depthBufferSize(),
+        EGL_STENCIL_SIZE, _glConfiguration.stencilBufferSize(),
+        EGL_RENDERABLE_TYPE, EGL_OPENGL_ES2_BIT,
+        EGL_NONE
+    };
+    EGLint configCount;
+    EGLConfig config;
+    if (!eglChooseConfig(_display, configAttributes, &config, 1, &configCount)) {
+        Error() << "Platform::AndroidApplication::tryCreate(): cannot choose EGL config:"
+            << Implementation::eglErrorString(eglGetError());
+        return false;
+    }
+
+    /* Resize native window and match it to the selected format */
+    EGLint format;
+    CORRADE_INTERNAL_ASSERT_OUTPUT(eglGetConfigAttrib(_display, config, EGL_NATIVE_VISUAL_ID, &format));
+    ANativeWindow_setBuffersGeometry(_state->window,
+        configuration.size().isZero() ? 0 : configuration.size().x(),
+        configuration.size().isZero() ? 0 : configuration.size().y(), format);
+
+    /* Create surface and context */
+    if (!(_surface = eglCreateWindowSurface(_display, config, _state->window, nullptr))) {
+        Error() << "Platform::AndroidApplication::tryCreate(): cannot create EGL window surface:"
+            << Implementation::eglErrorString(eglGetError());
+        return false;
+    }
+
+    /* Make the context current */
+    CORRADE_INTERNAL_ASSERT_OUTPUT(eglMakeCurrent(_display, _surface, _surface, _glContext));
+    return true;
+}
+
+void AndroidApplication::pauseContext() {
+    /* Pause the context current */
+    CORRADE_INTERNAL_ASSERT_OUTPUT(eglMakeCurrent(_display, nullptr, nullptr, nullptr));
```

Comment:
Would this work if you kept the context current and just made it bound to no surface at all?

```suggestion
    /* Keep the GL context current but without any surface */
    CORRADE_INTERNAL_ASSERT_OUTPUT(eglMakeCurrent(_display, EGL_NO_SURFACE, EGL_NO_SURFACE, _glContext));
```

Because I'm wondering if it's guaranteed that no app code gets executed between `APP_CMD_LOST_FOCUS` and `APP_CMD_INIT_WINDOW`. And if it does, it'd be probably good to have the GL context current.

Reply:
The suggested change, at least from my side, seems to work fine.

Reply:
> I'm wondering if it's guaranteed that no app code gets executed between APP_CMD_LOST_FOCUS and APP_CMD_INIT_WINDOW

I'm not sure about there being a guarantee, but it's one of the reasons I added `_application_has_valid_surface` - such that the user has some info to branch on.